apstools/plans/alignment.py

```diff
@@ -213,10 +217,133 @@ def peak_analysis():
+        y_data : A numpy array of the values on the y_axis
+
+        Returns
+        -------
+        guess : dict
+            A dictionary containing the guessed parameters 'low_y_data', 'high_y_data', 'width', and 'midpoint'.
+        """
+
+        # Sort data to make finding the mid-point easier and to assist in other estimations
+        y_data_sorted = np.sort(y_data)
+        x_data_sorted = np.sort(x_data)
+
+        # Estimate low and high as the first and last elements (assuming sorted data)
+        low_y_data = np.min(y_data_sorted)
+        high_y_data = np.max(y_data_sorted)
+
+        low_x_data = np.min(x_data_sorted)
+        high_x_data = np.max(x_data_sorted)
+
+        # Estimate wid as a fraction of the range. This is very arbitrary and might need tuning!
+        width = (
+            (high_x_data - low_x_data) / 10
+        )  # This is a guess and might need adjustment based on your data's characteristics
+
+        # Estimate the midpoint of the x values
+        midpoint = x_data[int(len(x_data) / 2)]
+
+        return [low_y_data, high_y_data, width, midpoint]
+
+    def erf_model(x, low, high, width, midpoint):
+        """
+        Create error function for fitting and simulation
+
+        Parameters
+        ----------
+        x		:	input upon which error function is evaluated
+        low		: 	min value of error function
+        high	: 	max value of error function
+        width  	:	"spread" of error function transition region
+        midpoint:	location of error function's "center"
+        """
+        return (high - low) * 0.5 * (1 - erf((x - midpoint) / width)) + low
+
+    if not isinstance(detectors, (tuple, list)):
+        detectors = [detectors]
+
+    _md = dict(purpose="edge_align")
+    _md.update(md or {})
+
+    uid = yield from bp.scan(detectors, mover, start, end, points, md=_md)
+    cat = cat or utils.getCatalog()
+    run = cat[uid]  # return uids
+    ds = run.primary.read()
+
+    x = ds["mover"]
+    y = ds["noisy"]
+
+    initial_guess = guess_erf_params(x, y)
+
+    ##################### TODO: Find a better way to check if the signal is a good one. The below method gives false positives on ocassion
+    # try:
+    #     with warnings.catch_warnings():
+    #         warnings.filterwarnings(
+    #             "error", "Covariance of the parameters could not be estimated"
+    #         )
+    #         popt, pcov = curve_fit(erf_model, x, y, p0=initial_guess)
+
+    #         perr = np.sqrt(np.diag(pcov))
+
+    #         print(f"guess:{initial_guess}")
+    #         print(f"perr:{perr}")
+    #         print(f"popt:{popt}")
+
+    #         yield from bps.mv(mover, popt[3])
+
+    # except UserWarning as e:
+    #     print(f"Warning caught as exception: {e}")
+    #     print("\n Check to see if signal is linear.\n")
+    ###################
+
+    popt, pcov = curve_fit(erf_model, x, y, p0=initial_guess)
+    yield from bps.mv(mover, popt[3])
```

Comment:
Only call `bps.mv(...)` if the curve fit results are acceptable.  Looking for an `if` here.